component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthenticator.java

```diff
@@ -106,6 +110,12 @@ private void readServerConfig(){
     public boolean authenticateClient(HttpServletRequest httpServletRequest, Map<String, List> bodyParameters,
                                       OAuthClientAuthnContext oAuthClientAuthnContext) throws OAuthClientAuthnException {
 
+        if (Boolean.parseBoolean(isFapiApplication(getClientId(httpServletRequest, bodyParameters,
```

Comment:
Wondering if it's better to change method name as "isFAPIApplication", wdyt?

Reply:
This method is added to this PR temporarily and a separate method will be used for this purpose which would be maintained in a common location as there will be multiple instances in which this method needs to be accessed for validation purposes.

Reply:
Fixed in c8efc7a5d86d1df217e9bdcab4167db2d8a20ab5